### ISSUE TITLE
Type mismatch on a clean project (PluginRegistry instead of FlutterEngine)

### ISSUE THREAD

**Issue by paolorotolo** [bkonyi/FlutterGeofencing#43]
Following the steps in `REAMDE` and including the plugin from git, I'm unable to compile the sources in a clean project.

It only compiles if I replace 
```kotlin
    override fun registerWith(registry: PluginRegistry) {
        GeneratedPluginRegistrant.registerWith(registry);
    }
```
with
```kotlin
    override fun registerWith(registry: PluginRegistry) {
        registry.registrarFor("io.flutter.plugins.geofencing.GeofencingPlugin")
    }
```
in `Application.kt`

Log:
```
Application.kt: (17, 48): Type mismatch: inferred type is PluginRegistry but FlutterEngine was expected
```

**Comment by bkonyi** [comment]
Yes, if you're using a recently created Flutter project the Android implementation is slightly different due to an API change. This needs to be updated still.

**Comment by SungKyung-Kim** [comment]
> Yes, if you're using a recently created Flutter project the Android implementation is slightly different due to an API change. This needs to be updated still.

I eagerly hope that the new Android Plugins API of Flutter 1.12 will be applied.
Cheer up Ben.

**Comment by paolorotolo** [comment]
Yeah, you might want to add this tip in README while we wait for a fix for new users.
If it's ok for you I can make a quick PR.

**Comment by mateenkiani** [comment]
If someone got it working please mention the solution I am still having this issue using the latest version of plugin and flutter version is 1.22.5 stable.

**Comment by paolorotolo** [comment]
There's a possible workaround in the first post. See if it still works.

**Comment by mateenkiani** [comment]
> There's a possible workaround in the first post. See if it still works.

I have tried that but the application crashes on launch
